### PR TITLE
Weather calls Alpaca API

### PR DESCRIPTION
**Pull request by gauntc**
Before the controller was simply reading a text file, but now it has been modified to call the Alpaca API on an observatory's weather sensors.  It makes several concurrent calls for each weather metric and aggregates them into a weather status object.  Also included are a set of tests that mock the Alpaca call.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,30 +1,53 @@
...
+        if (result.data.ErrorNumber === 0) {
+          weather[weatherProperty] = result.data.Value;
+        } else {
+          console.error('\t\tError occurred in "' + weatherProperty.toLowerCase() + '" call to weather equipment: Error ' +
+            result.data.ErrorNumber + ' - ' + result.data.ErrorMessage);
+        }
+      }
+    })
+    .catch((error) => {
+      console.error(error);
```
Comment (blittle) [inline index.js:44]: I should expose a logging mechanism that you can call so we can keep logs consistent and easily recorded

File: index.js [hunk 2]
```diff
@@ -1,30 +1,53 @@
-const fs = require('fs');
+const axios = require('axios');
+var clientTransactionId = 1;
```
Comment (blittle) [inline index.js]: nit picky, but I'd `let` this instead of `var`

File: index.js [hunk 3]
```diff
@@ -1,30 +1,53 @@
...
-
-  return lines.filter(line => line[0]).reduce(
-    (output, line) => ({
-      ...output,
-      [line[0]]: line[1],
-    }),
-    {},
-  );
+  let weatherProperties = Object.getOwnPropertyNames(weather);
+  await axios.all(weatherProperties.map(prop => getStat(config, prop.toLowerCase())))
```
Comment (blittle) [inline index.js]: Might there be a better way to do this? If any one of the requests fail, then `.then` will never resolve. Is that desirable?  Or would it be better to resolve some of the weather values, even if one or two fail?

Comment (gauntc) [reply index.js]: Thanks for catching that!  My intention was indeed for the weather object to aggregated regardless of how many of the individual calls failed.  I fixed this by adding a catch block on the axios call within getStat.  The catch block returns null so getStat should always return some value.  The aggregation logic then checks for the erroneous null value and reacts appropriately.

**Review by blittle — APPROVED** [review]
I think it looks good. Let's address better logging in a separate PR

**Pull request merged**